vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -204,6 +211,12 @@ def select_gemm_impl(
             f"{self.__class__.__name__} must select appropriate gemm "
             "implementation based on the prepare_finalize")
 
+    def maybe_swap_experts_impl(
+        self,
+        moe_parallel_config: FusedMoEParallelConfig,
+    ):
```

Comment:
Would be good to leave a docstring on the purpose of this function and when to implement it